aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/service/AWSS3StorageService.kt

```diff
@@ -85,6 +88,30 @@ internal class AWSS3StorageService(
         return URL(presignUrlRequest.url.toString())
     }
 
+    /**
+     * Validate if S3 object exists for the given key.
+     * Throws StorageException if NoSuchKey S3 client exception is caught.
+     * @param serviceKey S3 service key
+     */
+    override fun validateObjectExists(serviceKey: String) {
+        try {
+            runBlocking {
+                s3Client.headObject(
+                    HeadObjectRequest {
+                        bucket = s3BucketName
+                        key = serviceKey
+                    }
+                )
+            }
+        } catch (ex: NotFound) {
```

Comment:
if we're catching a specific exception, should we also have a catch base exception below this? If S3 ever increases its exception types, then the second catch on the base type should catch it

Reply:
See my comments that we should not catch here but instead in the operation itself. We can specifically catch on this message to provide better error messaging and let the others fall through to more standard messaging.

Reply:
exception handling moved to operation per @tylerjroach suggestions.